sphinx/ext/autodoc/__init__.py

```diff
@@ -1370,7 +1384,29 @@ def format_signature(self, **kwargs: Any) -> str:
         if self.doc_as_attr:
             return ''
 
-        return super().format_signature(**kwargs)
+        overloaded = False
+        qualname = None
+        cls, method_name, _ = self._get_signature()
+        # TODO: recreate analyzer for the module of class (To be clear, owner of the method)
```

Comment:
`self.analyzer` only loads overloads mapping for the module of current class. But the `__init__`, `__new__` or `__call__` method might be implemented at other module. So we need to recreate an analyzer for the module where the method actually implemented.